### PR TITLE
DEV-2263 | pretty print read only json fields in ContributionAdmin

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### What's this PR do?

- Creates a new function called `prettify_json_field` [as documented in this helpful blog post](https://daniel.feldroy.com/posts/pretty-formatting-json-django-admin) to pretty print read-only JSON ModelAdmin fields
- Uses this new function to pretty print bad_actor_response, provider_payment_method_details, and payment_provider_data

#### Why are we doing this? How does it help us?

Allow admin users to more easily decipher JSON data when looking at contribution data.

#### How should this be manually tested? Please include detailed step-by-step instructions.

Look at a contribution in Django admin in review app. The bad_actor_response, provider_payment_method_details, and payment_provider_data fields should each be nicely indented, and much more readable than before.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

No. There's not a great way to test formatting of JSON fields in the UI. If anyone has ideas about a good, valuable way to test, I'm all ears.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

[DEV-2263](https://news-revenue-hub.atlassian.net/browse/DEV-2263)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No

### CODE REVIEW

File: apps/common/admin.py [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+from pygments.lexers import JsonLexer
 
 
 class RevEngineBaseAdmin(ModelAdmin):
     formfield_overrides = {
         models.JSONField: {"widget": JSONEditorWidget},
     }
+
+
+def prettify_json_field(raw_value, indent=2):  # pragma: no cover
```
Comment (njharman) [inline apps/common/admin.py:19]: > There's not a great way to test formatting of JSON fields in the UI. If anyone has ideas about a good, valuable way to test, I'm all ears.

I would add tests to this function that verify some inputs return str? [I'm not sure what mark_safe does technically]; 
 empty, str(input) != output ie some formatting occured.  Maybe print out the result from formatting a small bit of json and put that in a test (I probably wouldn't bother with that (until/unless a bug occurred), I tend to trust 3rd party dumps/HtmlFormatter and am more interested that unit tests to fail on future refactors like function args/inputs output type.

File: apps/contributions/admin.py [hunk 2]
```diff
@@ -201,3 +201,21 @@ def _generate_stripe_connect_link(self, slug, provider_id, stripe_account_id):
...
+        """Render payment_provider_data field with pretty formatting"""
+        return prettify_json_field(instance.payment_provider_data)
+
+    payment_provider_data_pretty.short_description = "Payment provider data"
+
+    def provider_payment_method_details_pretty(self, instance):
+        """Render provider_payment_method_details field with pretty formatting"""
+        return prettify_json_field(instance.provider_payment_method_details)
+
+    provider_payment_method_details_pretty.short_description = "Provider payment method details"
```
Comment (njharman) [inline apps/contributions/admin.py:221]: I would add tests that call these methods (probably one test parameterized with attribute; bad_actor_response, payment_provider_data, provider_payment_method_details). Or, at least mark them # no qa if it's too hard to instantiate Admin instance

**Review by njharman — CHANGES_REQUESTED** [review]
looks good, answered how I would add unit tests.  Although, I wouldn't argue if your opinion is they're not good enough ROI.

File: apps/common/admin.py (hunk 1 above)
Comment (benjaminEwhite) [reply apps/common/admin.py:19]: Added minimal test in this commit: f3aac89572fb

File: apps/contributions/admin.py (hunk 2 above)
Comment (benjaminEwhite) [reply apps/contributions/admin.py:221]: Added minimal tests in this commit: f3aac89572fb

**Review by njharman — APPROVED** [review]

**Review by x110dc — APPROVED** [review]

**Pull request merged**